File: rb/features.rb
# Autoscrape SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module AutoscrapeFeatures
  def self.make_feature(name)
    case name
    when "base"
      AutoscrapeBaseFeature.new
    when "ratelimit"
      AutoscrapeRatelimitFeature.new
    when "retry"
      AutoscrapeRetryFeature.new
    when "test"
      AutoscrapeTestFeature.new
    when "timeout"
      AutoscrapeTimeoutFeature.new
    else
      AutoscrapeBaseFeature.new
    end
  end
end

File: py/features.py
# Autoscrape SDK feature factory

from feature.base_feature import AutoscrapeBaseFeature
from feature.test_feature import AutoscrapeTestFeature


def _make_feature(name):
    features = {
        "base": lambda: AutoscrapeBaseFeature(),
        "test": lambda: AutoscrapeTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
